python/modules/IcePy/ValueFactoryManager.cpp

```diff
@@ -52,12 +49,6 @@ IcePy::ValueFactoryManager::create()
 
 IcePy::ValueFactoryManager::ValueFactoryManager() : _defaultFactory{make_shared<DefaultValueFactory>()} {}
 
-IcePy::ValueFactoryManager::~ValueFactoryManager()
-{
-    AdoptThread adoptThread; // Ensure the current thread is able to call into Python.
```

Comment:
I think it's indeed ok to remove this code because ValueFactoryManager refcounts are only owned by:
 - the communicator (see destroy)
 - the Python application